platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/event/ChangeEvents.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Manages the listening and handling of change events.
+ */
+class ChangeEventManager {
+
+	public setListenToChangeEvent(node: HTMLElement, listen: boolean) {
+		if(listen) {
+			node.onchange = event => this.handleChangeEvent(event);
+		} else {
+			node.onchange = null;
+		}
+	}
+
+	private handleChangeEvent(event: Event) {
+		let node = event.currentTarget as HTMLElement;
+
+		// skip native change events for auto-complete inputs
+		if(event.isTrusted && AUTO_COMPLETE_ENGINE.isEnabledForInput(node)) {
+			return;
+		}
+
+		// only send or delegate event if value really changed
+		if(VALUE_NODE_MAP.isValueChanged(node)) {
+			sendOrDelegateEvent(node, event, event.type);
+		}
+	}
+}
+
+const CHANGE_EVENT_MANAGER = new ChangeEventManager();
```

Comment:
Should we somehow advertize this as the singleton handle to this class? We could add a comment here, or above the class.
Applies to other classes as well.

Reply:
my plan is to gather all global variables in the instance of `AjaxEngine.ts`.

Reply:
Sounds reasonable